packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -30,152 +31,58 @@ export type CreateViewResult = {
-    // Stop render was called on this stream after we had started rendering. We will dispose this view and do not
-    // put the view into the state.
-    _logStreamEvent(EventNames.CREATED_STREAM_STOPPING, streamLogInfo);
-    renderer.dispose();
-    internalContext.setRemoteRenderInfo(
-      callId,
-      participantKey,
-      streamId,
-      refreshedRenderInfo.stream,
-      'NotRendered',
-      undefined
-    );
-    context.setRemoteVideoStreamRendererView(callId, participantKey, streamId, undefined);
-    return;
+  if (participantId) {
+    streamEventType = 'createViewRemote';
+  } else if (callId) {
+    streamEventType = 'createViewLocal';
+  } else {
+    // TODO update for when unparented view.
+    throw new Error('unparented createView not implemented yet here');
+    streamEventType = 'createViewUnparented';
   }
 
-  // Else the stream still exists and status is not telling us to stop rendering. Complete the render process by
-  // updating the state.
-  internalContext.setRemoteRenderInfo(
-    callId,
-    participantKey,
-    streamId,
-    refreshedRenderInfo.stream,
-    'Rendered',
-    renderer
-  );
-  context.setRemoteVideoStreamRendererView(
-    callId,
-    participantKey,
-    streamId,
-    convertFromSDKToDeclarativeVideoStreamRendererView(view)
-  );
-  _logStreamEvent(EventNames.VIEW_RENDER_SUCCEED, streamLogInfo);
+  const streamType = stream?.mediaStreamType;
+  const streamId = (stream as RemoteVideoStream).id;
 
-  return {
-    renderer,
-    view
-  };
-}
-
-async function createViewLocalVideo(
-  context: CallContext,
-  internalContext: InternalCallContext,
-  callId: string,
-  options?: CreateViewOptions
-): Promise<CreateViewResult | undefined> {
-  // Render LocalVideoStream that is part of a Call
-  const renderInfo = internalContext.getLocalRenderInfo(callId);
+  // we want to check to see if there is a participantId this will tell us whether its a local stream or a remote one.
+  const participantKey =
+    streamEventType === 'createViewRemote' && participantId
+      ? typeof participantId === 'string'
+        ? participantId
+        : toFlatCommunicationIdentifier(participantId)
+      : undefined;
 
-  const streamType = renderInfo?.stream.mediaStreamType;
+  const streamLogInfo = { callId, participantKey, streamId, streamType, streamEventType };
 
-  const streamLogInfo = { callId, streamType, streamEventType: 'createViewLocal' };
+  // make different logging announcement based on whether or not we are starting a local or remote
+  _logStreamEvent(EventNames.CREATING_VIEW, streamLogInfo);
 
-  _logStreamEvent(EventNames.START_STREAM_RENDERING, streamLogInfo);
+  // if we have a participant Id and a stream get the remote info, else get the local render info from state.
+  const renderInfo =
+    streamEventType === 'createViewRemote' && participantKey
+      ? internalContext.getRemoteRenderInfoForParticipant(callId, participantKey, streamId)
```

Comment:
not for this pr, this pr already does loads of goodness!
But we should make the `internalContext.set/get` interfaces agnostic of remote vs local to avoid all these ternaries we have throughout.